examples/plot_glm_retinal_ganglia.py

```diff
@@ -0,0 +1,264 @@
+                  ' spike counts')
+axes[0].set_ylabel('Stimulation Intensity')
+
+tspplot = spike_time[(spike_time >= t_sample.min()) &
+                     (spike_time < t_sample.max())]
+markerline, _, _ = axes[1].stem(spike_time[sample_idx],
+                                [1] * len(spike_time[sample_idx]))
+markerline.set_markerfacecolor('none')
+axes[1].set_xlim([0., tmax])
+axes[1].set_ylim([0, 2])
+axes[1].set_ylabel('Spikes')
+
+markerline, _, _ = axes[2].stem(t_sample, spikes_binned[sample_idx])
+markerline.set_markerfacecolor('none')
+axes[2].set_xlim([0., tmax])
+axes[2].set_xlabel('Time (s)')
+axes[2].set_ylabel('Binned Spike counts')
+plt.show()
+
+########################################################
+#
+# We can use ``scipy``'s function ``hankel`` to make our design matrix.
+# Design matrix :math:`X` can be created using the stimulation and its history.
+# Later in the tutorial, we will also incorporate spikes history into our
+# design matrix.
+
+n_t_filt = 25  # tweak this to see different results
+stim_padded = np.pad(stim, (n_t_filt - 1, 0))
+Xdsgn = hankel(stim_padded[0: -n_t_filt + 1], stim[-n_t_filt:])
+
+plt.figure()
+plt.imshow(Xdsgn[:50, :],
+           cmap='binary',
+           aspect='auto',
+           interpolation='nearest')
+plt.xlabel('lags before spike time')
+plt.ylabel('time bin of response')
+plt.title('Sample first 50 rows of design'
+          ' matrix created using Hankel')
+plt.show()
+
+########################################################
+# **Fitting and predicting with a linear-Gaussian GLM**
+#
+# For a general linear model, an observed spikes can be
+# thought of an underlying parameter
+# :math:`\beta_0, \beta` that control the spiking:
+# :math:`y = (\beta_0 + X \beta) \ + \epsilon` and
+# :math:`y \sim \text{Poiss}(\beta_0 + X \beta)` where
+# :math:`X` is the stimulation and history of stimulation
+#
+# You can simply use linear Gaussian GLM with no regularization
+# to fit as follows
+
+glm_lgGLM = GLM(distr='gaussian', reg_lambda=0.0)
+glm_lgGLM.fit(Xdsgn, spikes_binned)
+
+# predict spike counts
+spikes_pred_lgGLM_offset = glm_lgGLM.predict(Xdsgn)
+
+########################################################
+# **Fitting and predicting with a Gaussian GLM**
+#
+# We can also assume that their is a non-linear function governing
+# the underlying the firing patterns. Concreately, we can write down as
+# :math:`y = q(\beta_0 + X \beta) + \epsilon`, and
+# :math:`y \sim \text{Poiss}(\beta_0 + X \beta)`.
+# Here we use :math:`exp()` as the inverse link function :math:`q`.
+# We can use the :class:`GLM` class to predict the parameters.
+
+# create possion GLM instance
+glm_poisson = GLM(distr='poisson',
+                  alpha=0.05,
+                  learning_rate=1.0,
+                  score_metric='pseudo_R2',
+                  reg_lambda=1e-7)
+glm_poisson.fit(Xdsgn, spikes_binned)
+
+# predict spike counts
+spikes_pred_poissonGLM = glm_poisson.predict(Xdsgn)
```

Comment:
```suggestion
ypred_poisson = glm_poisson.predict(Xdsgn)
```

GLM is common to all three methods, so it seems a bit superfluous in the name